pkg/kv/kvserver/multi_sst_writer.go

```diff
@@ -100,6 +100,78 @@ func newMultiSSTWriter(
 	return msstw, nil
 }
 
+func (msstw *multiSSTWriter) readOneToBatch(
```

Comment:
Hard to say in advance whether this belongs to `multiSSTWriter`. Logically, this type could be agnostic of where the key/values come from (`batchReader` or something else e.g. inserted manually) and be only concerned with how to write them. Right now the API becomes a mix of low-level `msstw.Put<something>` and high-level "feed a whole batch", and it's easy to misuse.

No strong suggestion on how to do it differently though, it's probably fine as is since this code is coupled anyway and it's quite nice to have it all in one place.

Is there some expectation to capture in a comment, e.g. on the order in which batches are fed into here? Can there be more than one? Does moving this method here make `Put<something>` methods "internal" and they no longer should be intended for use?

Reply:
I was able to make all the Put methods internal.